src/Model/ManagerInterface.php

```diff
@@ -17,20 +17,22 @@
 
 /**
  * @author Sylvain Deloux <sylvain.deloux@ekino.com>
+ *
+ * @template-covariant T of object
```

Comment:
Where can I find some documentation on `template-covariant`? I don't know what it means.

Reply:
I'm not sure if there is an offical documentation :/ phpstan doesn't have one.

Maybe `@template T of object` is more precise 🤔

Reply:
Even the psalm documentation does not say anything about `@template-covariant`, although they are using them in some test files
https://github.com/vimeo/psalm/blob/master/docs/annotating_code/templated_annotations.md

Reply:
@Ocramius maybe you can help us here?

Reply:
Looks like it was thought in https://github.com/vimeo/psalm/issues/1603

Reply:
OK so it's about allowing instances of types that extend `T`? Does it make sense here?

Reply:
> OK so it's about allowing instances of types that extend `T`? Does it make sense here?

Yes, because `T` will be a specific object (e.g. User or BlogPost)

Reply:
My question is: could a manager serve instances of class that extend from `User` (`PremiumUser`, `RegularUser`, whatever…)?

Reply:
> My question is: could a manager serve instances of class that extend from `User` (`PremiumUser`, `RegularUser`, whatever…)?

AFAIK yes, because you are not restricted only to a specific class